ee/clickhouse/models/test/test_cohort.py

```diff
@@ -654,3 +655,66 @@ def test_static_cohort_precalculated(self):
                     reindent=True,
                 ),
             )
+
+    def test_cohortpeople_with_valid_other_cohort_filter(self):
+        p1 = Person.objects.create(team_id=self.team.pk, distinct_ids=["1"], properties={"foo": "bar"},)
+        p2 = Person.objects.create(team_id=self.team.pk, distinct_ids=["2"], properties={"foo": "non"},)
+
+        cohort0: Cohort = Cohort.objects.create(
+            team=self.team, groups=[{"properties": {"foo": "bar"}}], name="cohort0",
+        )
+        cohort0.calculate_people_ch()
+
+        cohort1: Cohort = Cohort.objects.create(
+            team=self.team,
+            groups=[{"properties": [{"key": "id", "type": "cohort", "value": cohort0.id}]}],
+            name="cohort1",
+        )
+
+        cohort1.calculate_people_ch()
+
+        count_result = cast(
+            List[Tuple[int]],
+            sync_execute(
+                "SELECT count(person_id) FROM cohortpeople where cohort_id = %(cohort_id)s", {"cohort_id": cohort1.pk}
+            ),
+        )[0][0]
+        self.assertEqual(count_result, 1)
+
+    def test_cohortpeople_with_nonexistent_other_cohort_filter(self):
+        p1 = Person.objects.create(team_id=self.team.pk, distinct_ids=["1"], properties={"foo": "bar"},)
+        p2 = Person.objects.create(team_id=self.team.pk, distinct_ids=["2"], properties={"foo": "non"},)
+
+        cohort1: Cohort = Cohort.objects.create(
+            team=self.team, groups=[{"properties": [{"key": "id", "type": "cohort", "value": 666}]}], name="cohort1",
+        )
+
+        cohort1.calculate_people_ch()
+
+        count_result = cast(
+            List[Tuple[int]],
+            sync_execute(
+                "SELECT count(person_id) FROM cohortpeople where cohort_id = %(cohort_id)s", {"cohort_id": cohort1.pk}
+            ),
+        )[0][0]
+        self.assertEqual(count_result, 0)
+
+    def test_cohortpeople_with_cyclic_cohort_filter(self):
+        p1 = Person.objects.create(team_id=self.team.pk, distinct_ids=["1"], properties={"foo": "bar"},)
+        p2 = Person.objects.create(team_id=self.team.pk, distinct_ids=["2"], properties={"foo": "non"},)
+
+        cohort1: Cohort = Cohort.objects.create(
+            team=self.team, groups=[], name="cohort1",
+        )
+        cohort1.groups = [{"properties": [{"key": "id", "type": "cohort", "value": cohort1.id}]}]
+        cohort1.save()
+
+        cohort1.calculate_people_ch()
+
+        count_result = cast(
```

Comment:
I don't think we need a cast here or anywhere above - just adds noise.

Reply:
Well, Pylance doesn't like this, but Mypy is okay with it, so :+1: